src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -86,7 +86,7 @@ internal ZipArchiveEntry(ZipArchive archive, ZipCentralDirectoryFileHeader cd)
 
             _fileComment = cd.FileComment;
 
-            _compressionLevel = null;
+            _compressionLevel = MapCompressionLevel(_generalPurposeBitFlag);
```

Comment:
I think the better option here might be to eliminate the null check. The only time that `_compressionLevel` is going to be null is when creating a new ZipArchiveEntry. This sets `_compressionLevel` to null, then sets the bit flags to zero in `MapDeflateCompressionOption`. `GetDataCompressor` will then interpret the null as `CompressionLevel.Optimal` implicitly.

If the same ZIP file is loaded again though, `MapCompressionLevel` will map the zero to `CompressionLevel.Optimal` explicitly. I think it'd be clearer to make `_compressionLevel` non-nullable, set it to `CompressionLevel.Optimal` explicitly in the constructor and remove the coalesce inside `GetDataCompressor`.

Doing this means that there's no extra check in `GetDataCompressor`'s default case for a test to reach. There's provably no behavioural change here to test because the current behaviour is effectively:
```csharp
_compressionLevel = null;
// ...
compressorStream = new DeflateStream(/**/, _compressionLevel ?? CompressionLevel.Optimal, /**/);
```
while the new behaviour would be:
```csharp
_compressionLevel = CompressionLevel.Optimal;
// ...
compressorStream = new DeflateStream(/**/, _compressionLevel, /**/);
```